Resource leak checker documentation

kelloggm opened the pull request.

mernst commented on docs/manual/resource-leak-checker.tex:
```diff
@@ -0,0 +1,301 @@
+(1) treats the \<@MustCall>
+obligation of $expr$ as \emph{satisfied},
+and (2) creates a fresh obligation to check.
+This is sound: the checker creates a new obligation for calls to
```
Please replace indefinite references with specific ones.  I'm not sure whether "this" refers to both 1&2 or just 2.  I have similar questions about "this check" and "the latter".

kelloggm commented on docs/manual/resource-leak-checker.tex:
```diff
@@ -0,0 +1,301 @@
+\<@CreatesObligation>,
+indicates for which expression an obligation is created.
+
+When checking a call to a method annotated as
```
The only requirements on a method with an `@CreatesObligation` annotation are:
* the target is in-scope and parseable
* the method does not override a method with a weaker `@CreatesObligation` annotation (including no such annotation)

In general, these requirements are pretty easy to fulfill. The requirement you're referring to in your question doesn't come from the `@CreatesObligation` annotation itself, but rather from re-assigning a field (which also creates a requirement to write an `@CreatesObligation` annotation, which I think explains your confusion).

I added a note to the text that explains those two conditions on where the annotation can be written, and extended the discussion generally. Let me know if this is still not clear.

kelloggm commented:
@mernst, I've addressed your PR comments. Can you take another look? Thanks!

kelloggm commented:
I've addressed the comments on this PR in the `resource-leak-checker-full` branch, which will be the basis for a PR to the CF proper. I'm closing this PR now; further review will take place on that future PR.
